oak-api/src/main/java/org/apache/jackrabbit/oak/api/jmx/QueryEngineSettingsMBean.java

```diff
@@ -86,6 +86,24 @@
 
     void setFastQuerySize(boolean fastQuerySize);
 
+    /**
+     * Whether Path restrictions are enabled while figuring out index plan
+     *
+     * @return true if enabled
+     */
+    boolean isEnablePathRestrictions();
+
+    /**
+     * Enable/Disable evaluating path restriction while figuring out which index should be used
+     *
+     * @param enablePathRestrictions enablePathRestrictions
+     */
+    @Description("Enable/Disable evaluating path restriction while figuring out which index should be used")
+    void setEnablePathRestrictions(
+            @Description("Whether path restrictions be evaluated during index plan evaluation")
```

Comment:
This description is seen in the JMX UI right? What about "Whether path restrictions of indexes (excludedPaths / includedPaths) are taken into account during query execution, for Lucene indexes. When enabled, only indexes are considered if the index path restriction is compatible with the query path restrictions. When disabled, only the queryPaths of the index is taken into account."

Reply:
The above can also be used for documentation (in oak-doc), together with some examples. Don't worry about the documentation right now, we can do that later.